### ISSUE TITLE
How could I change the code?

### ISSUE THREAD

**Issue by Chen65010445** [hysts/pytorch_image_classification#17]
![image](https://user-images.githubusercontent.com/28993670/90465064-2b12d600-e141-11ea-8d50-241ee739c88e.png)

When I trying to use self-changed evaluate.py to evaluate my own dataset this error always came first. Do you know how could I change the code? I already add this :
`transforms.ToTensor()`in evaluate.py and changed the code in dataset.py as `return self.transform(self.x[index]), self.transform(self.y[index])`
Is there any other way to eliminate the error?

**Comment by hysts** [comment]
Hi, @Chen65010445 

Ah, sorry about that. Currently, the code only works with CIFAR and MNIST.
I think the `test_loader` in [this code](https://github.com/hysts/pytorch_image_classification/blob/master/evaluate.py#L107) is a tuple of `train_loader` and `val_loader` in your case because of [this code](https://github.com/hysts/pytorch_image_classification/blob/7df3cd8b0e830824d610d6390590059e9cc3595a/pytorch_image_classification/datasets/datasets.py#L77-L85).
So I think you can change that part of the code appropriately to make it work.

**Comment by Chen65010445** [comment]
But I can finish the training with my own dataset in this code. So is there anything wrong I couldn't finish the prediction?

**Comment by hysts** [comment]
@Chen65010445 

Oh, sorry, I didn't notice your comment.

The reason you could train your model with your own dataset without errors is because the training code only uses a training dataset and a validation dataset and it doesn't need a test dataset.

Although you need a test dataset to run `evaluate.py`, [this code](https://github.com/hysts/pytorch_image_classification/blob/7df3cd8b0e830824d610d6390590059e9cc3595a/pytorch_image_classification/datasets/datasets.py#L77-L85) doesn't have the functionality to return `test_dataset`. And the reason it doesn't is because the ImageNet dataset doesn't have test labels. So, for your dataset, you need to modify the code to return your test dataset.
